File: 백준/Bronze/9506. 약수들의 합/약수들의 합.js
const cases = require('fs').readFileSync(0, 'utf-8').toString().trim().split('\n').map(Number);
let answer = '';

for (let n of cases) {
    if (n < 0) break;
    
    const factors = [];
    let sum = 0;
    
    for (let i = 1; i <= Math.floor(n / 2); i++) {
        if (n % i > 0) continue;
        
        factors.push(i);
        sum += i;
    }
    
    if (sum === n) {
        answer += `${n} = ${factors.join(' + ')}\n`;
    } else {
        answer += `${n} is NOT perfect.\n`;
    }
}

console.log(answer);

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.py
import sys

for line in sys.stdin:
    N = int(line.rstrip())
    factors = []
    
    if N < 0: break
    
    for i in range(N // 2 + 1):
        if N % (i + 1) == 0:
            factors.append(i + 1)
    
    if sum(factors) == N:
        print(f'{N} = {" + ".join(map(str, factors))}')
    else:
        print(f'{N} is NOT perfect.')